### PR TITLE
E2241 Heatgrid fixes and improvements

### PR DESCRIPTION
**Pull request by J-Sumer**
Please set the title of your pull request in the following format: `[Project id/Independent study/Issue fix/...]. project name`, eg., `E904. Integrating courses and assignments on Expertiza with Moodle`

E2241/Heatgrid fixes and improvements/Issue#1869

Changes made:

- Removed countofcomments attribute from vm_question_response_row.rb file.
- Added metric_hash attribute to store the metric values for each response in vm_question_response_row.rb.
- Iterate values in metric_hash in view_team.html.erb file to show the metric values in heat grid.
- Added a new method calculate_metrics, that calls all other methods for different metric calculation.

@EshwarCVS 
@gokul-koganti 
@J-Sumer 
@sbosenc

### CODE REVIEW

File: app/models/vm_question_response.rb [hunk 1]
```diff
@@ -161,14 +161,33 @@ def add_answer(answer)
     end
   end
 
+  def calculate_metrics
```
Comment (EshwarCVS) [inline app/models/vm_question_response.rb:164]: Need to add comments for this in the upcoming commits

File: config/logs.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (EshwarCVS) [inline config/logs.txt]: Do we really need the logs file to be pushed?

**Review by EshwarCVS — APPROVED** [review]

**Comment by gokul-koganti** [comment]
LGTM

**Review by gokul-koganti — APPROVED** [review]
Reviewed the changes. Code can be merged into the beta branch.

File: app/models/vm_question_response.rb (hunk 1 above)
Comment (sbosenc) [reply app/models/vm_question_response.rb:164]: Add comments for newly added method

File: app/models/vm_question_response.rb [hunk 3]
```diff
@@ -161,14 +161,33 @@ def add_answer(answer)
...
+    number_of_comments_greater_than_10_words
+  end
+
   def number_of_comments_greater_than_10_words
```
Comment (sbosenc) [inline app/models/vm_question_response.rb:168]: Add comment describing what the method is for

File: config/logs.txt (hunk 2 above)
Comment (sbosenc) [reply config/logs.txt]: I think logs.txt need not be in the PR

File: db/schema.rb [hunk 4]
```diff
@@ -11,230 +11,229 @@
 #
 # It's strongly recommended that you check this file into your version control system.
```
Comment (sbosenc) [inline db/schema.rb:13]: Do we need schemda.rb in the PR? We haven't changed anything.

**Review by sbosenc — CHANGES_REQUESTED** [review]
Please check the review comments before merging.

**Review by sbosenc — APPROVED** [review]

**Pull request merged**